components/threading/thread_viewer/thread_viewer.tsx

```diff
@@ -393,288 +198,36 @@ export default class ThreadViewer extends React.Component<Props, State> {
-                        autoHide={true}
-                        autoHideTimeout={500}
-                        autoHideDuration={500}
-                        renderThumbHorizontal={renderThumbHorizontal}
-                        renderThumbVertical={renderThumbVertical}
-                        autoHeightMax={this.state.postsContainerHeight}
-                        renderView={renderView}
-                        onScroll={this.handleScroll}
-                    >
-                        <div className='post-right__scroll'>
-                            <div
-                                role='application'
-                                aria-label={Utils.localizeMessage('accessibility.sections.rhsContent', 'message details complimentary region')}
-                                className='post-right__content a11y__region'
-                                data-a11y-sort-order='3'
-                                data-a11y-focus-child={true}
-                                data-a11y-order-reversed={true}
-                            >
-                                {!this.props.useRelativeTimestamp && !isFakeDeletedPost && <DateSeparator date={rootPostDay}/>}
-                                <RhsRootPost
-                                    post={selected}
-                                    commentCount={postsLength}
-                                    teamId={this.props.channel!.team_id}
-                                    currentUserId={this.props.currentUserId}
-                                    previewCollapsed={this.props.previewCollapsed}
-                                    previewEnabled={this.props.previewEnabled}
-                                    isBusy={this.state.isBusy}
-                                    handleCardClick={this.handleCardClick}
-                                    isLastPost={isRhsRootLastPost}
-                                    timestampProps={this.props.useRelativeTimestamp ? THREADING_TIME : undefined}
-                                />
-                                {!this.props.useRelativeTimestamp && isFakeDeletedPost && rootPostDay && <DateSeparator date={rootPostDay}/>}
-                                <div
-                                    ref={this.rhspostlistRef}
-                                    className='post-right-comments-container'
-                                >
-                                    {items}
-                                </div>
-                            </div>
-                            {createComment}
-                        </div>
-                    </Scrollbars>
+                <div className={classNames('ThreadViewer', this.props.className)}>
+                    <div style={{flex: '1 1 auto'}}>
+                        {this.props.selected && (
+                            <DeferredThreadViewerVirt
+                                key={`${this.props.selected.id}${this.props.highlightedPostId ?? ''}`}
```

Comment:
Since the `dynamic-virtualized-list` does not know the height of the whole list we cannot scroll without re-mounting on some occasions:

1. thread is open and we change selected post
2. thread is open and we highlight another post 
3. thread is open we select another thread which has new messages line.

That's why I've added this `key` prop.

PS: The items not rendered heights are not calculated, so the list does not know how far top they are, thus cannot scroll to those.